### PR TITLE
[enhancement] ban trailing underscore attributes to follow sklearn convention at class definition-level

### PR DESCRIPTION
**Pull request by icfaust**
## Description

Rules dictated by sklearn are violated but luckily not properly checked, this forces changes in sklearnex design to follow the rule closely.

https://scikit-learn.org/1.5/glossary.html#term-attributes

- Includes changes from #2102 (except SVM change) 
- Removes unnecessary class attributes from ```LogisticRegression```

---

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/intel/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.
- [x] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

- [x] I have measured performance for affected algorithms using [scikit-learn_bench](https://github.com/IntelPython/scikit-learn_bench) and provided at least summary table with measured data, if performance change is expected.
- [x] I have provided justification why performance has changed or why changes are not expected.
- [x] I have provided justification why quality metrics have changed or why changes are not expected.
- [x] I have extended benchmarking suite and provided corresponding scikit-learn_bench PR if new measurable functionality was introduced in this PR.

### CODE REVIEW

File: sklearnex/tests/test_common.py [hunk 1]
```diff
@@ -146,6 +146,24 @@ def wrap(*args, **kwargs):
...
+        if "preview" not in obj.__module__ and "daal4py" not in obj.__module__:
+            # propeties also occur in sklearn, especially in deprecations and are expected
+            # to error if queried and the estimator is not fitted
+            assert all(
+                [
+                    isinstance(getattr(obj, i), property)
+                    or (i.startswith("_") or not i.endswith("_"))
+                    for i in dir(obj)
+                ]
+            ), f"{name} contains class attributes which have a trailing underscore but no leading one"
```
Comment (samir-nasibli) [inline sklearnex/tests/test_common.py:164]:
```suggestion
            assert all(
                [
                    isinstance(getattr(obj, attr), property)
                    or (attr.startswith("_") or not attr.endswith("_"))
                    for attr in dir(obj)
                ]
            ), f"{name} contains class attributes which have a trailing underscore but no leading one"
```

**Review by samir-nasibli — APPROVED** [review]
